### PR TITLE
Adding IsGraspingObject to Physics Provider

### PR DESCRIPTION
**Pull request by rorygames**
## Summary

- Adds function to check if object is grasped via the physics provider.
- Adds example cubes to the example scene of object states and isgrasped functions.

## Contributor Tasks

_These tasks are for the merge request creator to tick off when creating a merge request._

- [x] Pair review with a member of the QA team.
- [x] Add any release testing considerations to the MR for the next release.
- [x] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.
- [x] If this MR closes a Jira issue, make sure the fix version on the JIRA issue is set to the correct one.

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

[Use emojis in review threads to communicate intent and help contributors.](https://github.com/ultraleap/UnityPlugin/blob/develop/CONTRIBUTING.md#review-threads)

- [ ] Code reviewed.
- [ ] Non-code assets e.g. Unity assets/scenes reviewed.
- [ ] Checked and agree with release testing considerations added to MR for the next release.

### CODE REVIEW

File: Packages/Tracking Preview/Examples~/PhysicsHands/Scripts/PhysicsGrasped.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+namespace Leap.Unity.Interaction.PhysicsHands.Example
+{
+    public class PhysicsGrasped : MonoBehaviour
```
Comment (MaxPalmer-UH) [inline Packages/Tracking Preview/Examples~/PhysicsHands/Scripts/PhysicsGrasped.cs:8]: Can we have comments on public classes and public interfaces please?

File: Packages/Tracking Preview/Examples~/PhysicsHands/Scripts/PhysicsState.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+namespace Leap.Unity.Interaction.PhysicsHands.Example
+{
+    public class PhysicsState : MonoBehaviour
```
Comment (MaxPalmer-UH) [inline Packages/Tracking Preview/Examples~/PhysicsHands/Scripts/PhysicsState.cs:8]: Once again, can we have comments please?

File: Packages/Tracking Preview/PhysicsHands/Runtime/PhysicsProvider.cs [hunk 3]
```diff
@@ -577,6 +578,22 @@ private void FindHandState(PhysicsHand hand)
             hand.SetGraspingMass(mass);
         }
 
+        public bool IsGraspingObject(Rigidbody rigid)
```
Comment (MaxPalmer-UH) [inline Packages/Tracking Preview/PhysicsHands/Runtime/PhysicsProvider.cs:581]: Please comment these properties and methods.

**Comment by MaxPalmer-UH** [comment]
Seeing a few issues. This error when running the scene:

image

Also the text on the cube doesn't go back to the not grasp state until my hand moves away from the cube - that's after I've let go.

https://user-images.githubusercontent.com/53301524/182159698-3bd1e385-9a7f-4d3d-9b2b-7cb301480ac3.mp4

**Comment by rorygames** [comment]
Fixed that bug, realised I messed up the logic check when simplifying it.

**Comment by MaxPalmer-UH** [comment]
Great. That's those issues sorted. Thanks.

**Review by MaxPalmer-UH — APPROVED** [review]

**Pull request merged**